client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -150,12 +149,13 @@ const Header: React.FunctionComponent = () => {
 	const hasPlaceholder =
 		!! siteTitle && ! recommendedDomainSuggestion && previousRecommendedDomain !== '';
 
-	const onDomainSelect = ( suggestion: DomainSuggestion | undefined ) => {
-		trackEventWithFlow( 'calypso_newsite_domain_select', {
-			domain_name: suggestion?.domain_name,
-		} );
-		setDomain( suggestion );
-	};
+	const showFullHeader = ! [
+		'IntentGathering',
+		'Domains',
+		'DomainsModal',
+		'Plans',
+		'PlansModal',
+	].includes( currentStep );
```

Comment:
I don't fully understand this. I think it's more intuitive to use positive conditions (as opposed to negative). 

```suggestion
	const showFullHeader = [
		'DesignSelection',
		'Style',
		'Signup',
		'Login',
		'CreateSite',
	].includes( currentStep );
```

Reply:
I wanted to maintain a list of exceptions 🤔 
Maybe it would be better to name it `const hideFullHeader`?

Reply:
Yup that also works.